WalletWasabi.Tests/UnitTests/TestWallet.cs

```diff
@@ -0,0 +1,108 @@
+using NBitcoin;
+using NBitcoin.Crypto;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.BitcoinCore.Rpc;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Crypto;
+using WalletWasabi.WabiSabi.Client;
+
+namespace WalletWasabi.Tests.UnitTests;
+
+public class TestWallet : IKeyChain, IDestinationProvider
+{
+	public TestWallet(string name, IRPCClient rpc)
+	{
+		Rpc = rpc;
+		Key = new Key(Hashes.SHA256(Encoding.ASCII.GetBytes(name)));
+	}
+
+	private IRPCClient Rpc { get; }
+	private List<Coin> Utxos { get; } = new ();
+	private Key Key { get; }
+	public PubKey PubKey => Key.PubKey;
+	public Script ScriptPubKey => Key.GetScriptPubKey(ScriptPubKeyType.Segwit);
+	public BitcoinAddress Address => Key.PubKey.GetAddress(ScriptPubKeyType.Segwit, Rpc.Network);
+
+	public async Task GenerateAsync(int blocks, CancellationToken cancellationToken)
+	{
+		var blockIds = await Rpc.GenerateToAddressAsync(blocks, Address, cancellationToken).ConfigureAwait(false);
+		foreach(var blockId in blockIds)
+		{
+			var block = await Rpc.GetBlockAsync(blockId, cancellationToken).ConfigureAwait(false);
+			var coinbaseTx = block.Transactions[0];
+			ScanTransaction(coinbaseTx);
+		}
+	}
+
+	public Transaction CreateSelfTransfer(FeeRate feeRate)
+	{
+		var biggestUtxo = Utxos.MaxBy(x => x.Amount);
+		var tx = Rpc.Network.CreateTransaction();
+		tx.Inputs.Add(biggestUtxo.Outpoint);
+		tx.Outputs.Add(biggestUtxo.Amount - feeRate.GetFee(82), Address);
```

Comment:
Would this work:

```suggestion
		tx.Outputs.Add(biggestUtxo.Amount - feeRate.GetFee(tx), Address);
```

or did you design it intentionally with tx virtual size = 82?

Reply:
We cannot do that because in that case the fee would be calculated before adding the output, resulting in a too small fee. I created a one input/two segwit outputs transaction and calculated the virtual size (82) that I am using here.

Reply:
I am OK with this as well, but why not `Address.ScriptPubKey.EstimateOutputVsize()`?